django/apps/registry.py

```diff
@@ -344,6 +346,76 @@ def clear_cache(self):
             for model in self.get_models(include_auto_created=True):
                 model._meta._expire_cache()
 
+    @staticmethod
+    def _model_key(model_or_name):
+        """
+        Returns an (app_label, model_name) tuple from a model or string.
+        """
+        if isinstance(model_or_name, six.string_types):
+            app_label, model_name = model_or_name.split(".")
+        else:
+            # It's actually a model class.
+            app_label = model_or_name._meta.app_label
+            model_name = model_or_name._meta.object_name
+        return app_label, model_name
+
+    def lazy_model_operation(self, function, *models_or_names, **kwargs):
```

Comment:
The introspection is just [this bit here](https://github.com/django/django/pull/3984/files?diff=unified#diff-c8a3e1d248f8f5b10af7ac5b1d8479f4L183) – it's used to provide a more helpful error message. Not critical.